### PR TITLE
feat(sdk-coin-sui): add switch delegation tx

### PR DESCRIPTION
**Pull request by alexvBitGo**
BOS-892

### CODE REVIEW

File: modules/sdk-coin-sui/src/lib/index.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import * as Utils from './utils';
 import * as Interface from './iface';
 
 export { KeyPair } from './keyPair';
+export { StakingBuilder } from './stakingBuilder';
+export { StakingTransaction } from './stakingTransaction';
```
Comment (linyah9912) [inline modules/sdk-coin-sui/src/lib/index.ts:6]: Missing the new transferTransaction

**Review by ewangbitgo — DISMISSED** [review]

**Review by linyah9912 — COMMENTED** [review]
Looking good! just need to export the missing transaction and good to go!

**Review by linyah9912 — DISMISSED** [review]

File: modules/sdk-coin-sui/src/lib/stakingTransaction.ts [hunk 2]
```diff
@@ -313,7 +358,30 @@ export class StakingTransaction extends Transaction<MoveCallTx> {
...
+   * @param {TransactionExplanation} explanationResult The transaction explanation to be completed
+   * @returns {TransactionExplanation}
+   */
+  explainSwitchDelegationTransaction(json: TxData, explanationResult: TransactionExplanation): TransactionExplanation {
+    return {
+      ...explanationResult,
+      fee: {
+        fee: this.suiTransaction.gasBudget.toString(),
+      },
+      type: TransactionType.StakingWithdraw,
```
Comment (jicli) [inline modules/sdk-coin-sui/src/lib/stakingTransaction.ts:380]:
```suggestion
      type: TransactionType.SwitchDelegation,
```

File: modules/sdk-coin-sui/src/lib/txnSchema.ts [hunk 3]
```diff
@@ -90,11 +91,26 @@ export const RequestWithdrawTransactionSchema = joi.object({
   arguments: RequestWithdrawArgumentsSchema.required(),
 });
 
+export const RequestSwitchTransactionSchema = joi.object({
```
Comment (jicli) [inline modules/sdk-coin-sui/src/lib/txnSchema.ts:94]: RequestSwitchDelegationTransactionSchema?

File: modules/sdk-coin-sui/src/lib/stakingTransaction.ts (hunk 2 above)
Comment (jicli) [reply modules/sdk-coin-sui/src/lib/stakingTransaction.ts:380]: or StakingSwitch ?

File: modules/sdk-coin-sui/src/lib/txnSchema.ts (hunk 3 above)
Comment (alexvBitGo) [reply modules/sdk-coin-sui/src/lib/txnSchema.ts:94]: yeah typo there

File: modules/sdk-coin-sui/src/lib/stakingTransaction.ts (hunk 2 above)
Comment (alexvBitGo) [reply modules/sdk-coin-sui/src/lib/stakingTransaction.ts]: done

**Review by linyah9912 — DISMISSED** [review]

**Review by linyah9912 — DISMISSED** [review]

**Review by jicli — APPROVED** [review]

**Review by linyah9912 — APPROVED** [review]

**Review by jicli — APPROVED** [review]

**Review by byhow — APPROVED** [review]

**Pull request merged**